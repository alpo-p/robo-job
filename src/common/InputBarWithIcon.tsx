import { Ionicons } from '@expo/vector-icons'
import { useTheme } from '@react-navigation/native'
import React from 'react'
import { TextInput, useColorScheme, View } from 'react-native'
import styles from './styles'
import { IoniconsIconNames } from './types/utilTypes'

interface P {
  value: string
  // eslint-disable-next-line no-unused-vars
  setValue: (v: string) => void
  iconName: IoniconsIconNames
  placeholderText?: string
}

// TODO: refactor
export default ({
  value,
  setValue,
  iconName,
  placeholderText = 'Search',
}: P) => {
  const { colors } = useTheme()
  const scheme = useColorScheme()
  // TODO: refactor bgColor and placeholder color to themes
  const bgColor =
    scheme === 'dark' ? styles.lightBgDarkMode : styles.lightBgLightMode
  const placeholderColor =
    scheme === 'dark' ? styles.lighterWhiteText : styles.lighterDarkText
  return (
    <View
      style={{
        flexDirection: 'row',
        backgroundColor: bgColor,
        margin: 16,
        borderRadius: 8,
        height: 8 * 4,
      }}
    >
      <Ionicons
        style={{
          padding: 8,
        }}
        name={iconName}
        color={colors.primary}
        size={15}
      />
      <TextInput
        style={{
          color: colors.text,
          width: '70%',
        }}
        value={value}
        onChangeText={setValue}
        placeholder={placeholderText}
        placeholderTextColor={placeholderColor}
      />
    </View>
  )
}
