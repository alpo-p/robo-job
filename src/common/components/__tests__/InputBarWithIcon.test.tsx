import React from 'react'
import renderer from 'react-test-renderer'
import { render, fireEvent } from '@testing-library/react-native'
import InputBarWithIcon from '../InputBarWithIcon'

describe('InputBarWithIcon', () => {
  const value = 'test'
  const setValue = jest.fn()

  it('should match snapshot', () => {
    const tree = renderer.create(
      <InputBarWithIcon iconName="flower" value={value} setValue={setValue} />,
    )
    expect(tree).toMatchSnapshot()
  })

  it('should call setValue when edited', () => {
    const placeholderText = 'test placeholder'

    const { getByPlaceholderText } = render(
      <InputBarWithIcon
        iconName="flower"
        value={value}
        setValue={setValue}
        placeholderText={placeholderText}
      />,
    )

    fireEvent.changeText(getByPlaceholderText(placeholderText), 'testi')
    expect(setValue).toHaveBeenCalledWith('testi')
  })
})
