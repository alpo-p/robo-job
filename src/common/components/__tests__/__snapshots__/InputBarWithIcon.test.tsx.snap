// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBarWithIcon should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f1f1f1",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 32,
        "margin": 16,
      },
      Object {},
    ]
  }
>
  <Text />
  <TextInput
    allowFontScaling={true}
    onChangeText={[MockFunction]}
    placeholder="Search"
    placeholderTextColor="rgba(28, 28, 30, 0.4)"
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "color": "rgb(28, 28, 30)",
        "width": "70%",
      }
    }
    underlineColorAndroid="transparent"
    value="test"
  />
</View>
`;
